File: 2025-9322-team04_finproject_java/src/main/java/Client_Java/player/model/WaitingRoomModel.java
package Client_Java.player.model;

public class WaitingRoomModel {
    private final String username;
    private int countdownTime;

    public WaitingRoomModel(String username, int initialCountdownTime) {
        this.username = username;
        this.countdownTime = initialCountdownTime;
    }

    public String getUsername() {
        return username;
    }

    public int getCountdownTime() {
        return countdownTime;
    }

    public void setCountdownTime(int countdownTime) {
        this.countdownTime = countdownTime;
    }

    public void decrementCountdownTime() {
        if (countdownTime > 0) {
            countdownTime--;
        }
    }
}

File: 2025-9322-team04_finproject_python/Client_Python/player/model/WaitingRoomModel.py
# waiting_room_model.py
from Client_Python.compilations import AdminService_idl
AdminService_idl._0_compilations.AdminService
from Client_Python.manager.PlayerCorbaManager import PlayerCorbaManager


class WaitingRoomModel:
    _instance = None  # Define class-level singleton holder
    def __init__(self, username):
        self.count_down_time = None
        self.username = username
        orb_args = ["-ORBInitRef", "NameService=corbaloc::localhost:4321/NameService"]
        self.manager = PlayerCorbaManager.get_instance(orb_args)

    def get_username(self):
        return self.username

    def request_to_join_game(self, username):
        stub = self.manager.get_game_service()
        stub.requestToJoinGame(username)

    def get_game_status(self, username):
        stub = self.manager.get_game_service()
        return stub.getGameProgressStatus(username)

    def get_current_word_mask(self, username):
        stub = self.manager.get_game_service()
        return stub.getCurrentWordMask(username)

    def get_count_down_time(self):
        stub = self.manager.get_admin_service()
        self.count_down_time = stub.getJoinTimeout()
        return self.count_down_time
